geometry/polygon_area.cpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file
+ * @brief Implementation of [Shoelace
+ * formula](https://en.wikipedia.org/wiki/Shoelace_formula).
+ *
+ * @details
+ * given a set of points of a simple polygon, the algorithm returns the area of
+ * the polygon
+ *
+ * @author [Karimi Zayan](https://github.com/waffleSheep)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <numeric>   /// for inner_product function
+#include <vector>    /// for vector data structure
+
+/**
+ * @namespace geometry
+ * @brief Geometry algorithms
+ */
+namespace geometry {
+/**
+ * @namespace polygon_area
+ * @brief polygon area functions
+ */
+namespace polygon_area {
+/**
+ * Point Struct
+ * @tparam for numeric type of the point
+ */
+template <typename T>
+struct Point {
+    T x, y;  /// x and y values for a point
+};
+
+/**
+ * Class for Polygon Area
+ * @tparam for numeric type of the points
+ */
+template <typename T>
+class Polygon_Area {
+    std::vector<Point<T>> points;  /// list of all points in the 2d space
+ public:
+    /**
+     * Constructor of a given class
+     *
+     * @param point_list list of all the points in the 2d space
+     */
+    explicit Polygon_Area(std::vector<Point<T>> point_list)
+        : points(std::move(point_list)){};
+    /**
+     * Calculates area of the simple polygon
+     * with the [Shoelace
+     * formula](https://en.wikipedia.org/wiki/Shoelace_formula)
+     *
+     * @returns the area of the simple polygon
+     */
+    T get_polygon_area() {
+        /**
+         * This lambda multiplies the x value of the
+         * first point by the y value of the second point
+         */
+        auto multiply_xy_of_two_points = [](const Point<T> &first_point,
+                                            const Point<T> &second_point) {
+            return first_point.x * second_point.y;
+        };
+        auto first_point = points.begin();             ///< first point iterator
+        auto last_point = std::prev(points.end(), 1);  ///< last point iterator
+        // first "shoelace" term, it is the sum of the x value of every point
+        // not including the last point multiplied by the y value of the point
+        // after
+        auto first_term =
+            std::inner_product(points.begin(), std::prev(points.end(), 1),
+                               std::next(points.begin(), 1), 0, std::plus<T>(),
+                               multiply_xy_of_two_points);
+        // second "shoelace" term, it is the sum of the y value of every point
+        // not including the last point multiplied by the x value of the point
+        // after
+        auto second_term = std::inner_product(
+            std::next(points.begin(), 1), points.end(), points.begin(), 0,
+            std::plus<T>(), multiply_xy_of_two_points);
```

Comment:
Please add one-line documentation explaining what does this variable does or is for.